doc/modules/linear_model.rst

```diff
@@ -848,25 +848,36 @@ Ridge Regression`_, see the example below.
 Logistic regression
 ===================
 
-Logistic regression, despite its name, is a linear model for classification
-rather than regression. Logistic regression is also known in the literature as
-logit regression, maximum-entropy classification (MaxEnt) or the log-linear
-classifier. In this model, the probabilities describing the possible outcomes
-of a single trial are modeled using a
-`logistic function <https://en.wikipedia.org/wiki/Logistic_function>`_.
+LogisticRegression, despite its name, is implemented as a linear model for
+classification rather than regression in terms of scikit-learn/ML nomenclature.
+Logistic regression is also known in the literature as logit regression,
+maximum-entropy classification (MaxEnt) or the log-linear classifier. In this
+model, the probabilities describing the possible outcomes of a single trial are
+modeled using a `logistic function
+<https://en.wikipedia.org/wiki/Logistic_function>`_.
 
 Logistic regression is implemented in :class:`LogisticRegression`.
 This implementation can fit binary, One-vs-Rest, or multinomial logistic
 regression with optional :math:`\ell_1`, :math:`\ell_2` or Elastic-Net
 regularization.
 
-.. note::
+.. note:: **Regularization**
 
     Regularization is applied by default, which is common in machine
     learning but not in statistics. Another advantage of regularization is
     that it improves numerical stability. No regularization amounts to
     setting C to a very high value.
 
+.. note:: **Logistic Regression as a special case of the Generalized Linear Regression Model (GLM)**
+
+    Logistic regression is a special case of
+    :ref:`generalized_linear_regression` with a Binomial/Bernoulli conditional
```

Comment:
I would suggest using this opportunity to change the name of the section from "Generalized Linear Regression" to "Generalized Linear Models" in the linear_model.rst.

Maybe we can also add the Bernoulli/Binomial distribution to said section, so that linking to them makes more sense.

What do you think?

Reply:
Yeah, I also feel it should be renamed to "Generalized Linear Models". It will make the docs more consistent and easy to understand. I will also add a section on Bernoulli and Binomial Distribution.